sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -560,6 +618,17 @@ case class Pmod(left: Expression, right: Expression) extends BinaryArithmetic {
     } else {
       s"${eval2.value} == 0"
     }
+    val divByZero = if (failOnError) {
```

Comment:
This is only used in the `if (!left.nullable && !right.nullable)` branch, we can move it to that branch.